kori/app/dao/store_product.py

```diff
@@ -37,6 +40,39 @@ def get_store_product(store_id: UUID, product_id: UUID) -> StoreProductSchema |
     return StoreProductSchema.from_orm(store_products[0]) if len(store_products) > 0 else None
 
 
+def get_products_by_stores(store_ids: list[UUID], product_name: str | None = None) -> list[ProductWithStock]:
+    session = db_connector.get_session()
+    current_timestamp = datetime.now()
+    result = (
+        session.query(StoreProduct.product_id, func.sum(StoreProduct.stock_available), ProductVersion)
+        .filter(StoreProduct.store_id.in_(store_ids))
+        .group_by(StoreProduct.product_id, ProductVersion.product_id, ProductVersion.version_id)
+        .join(ProductVersion, ProductVersion.product_id == StoreProduct.product_id)
```

Comment:
Why do you doing a join again if you already fetched all possible records with the CROSS JOIN above?